bin/node-template/runtime/src/template.rs

```diff
@@ -52,25 +56,52 @@ decl_module! {
 			Self::deposit_event(RawEvent::SomethingStored(something, who));
 			Ok(())
 		}
+
+		/// Another dummy entry point.
+		/// takes no parameters, attempts to increment storage value, and possibly throws an error
+		pub fn cause_error(origin) -> dispatch::DispatchResult {
+			// TODO: You only need this if you want to check it was signed.
```

Comment:
is this intentional? else better to say `NOTE:` as `TODO` can be confusing with leftover work from a PR

Reply:
You probably always need the check. However, if you want to check for non-signed you should use `ensure_none! `